Source/Scene/ShadowMap.js

```diff
@@ -307,110 +455,292 @@ define([
+                    color : ColorGeometryInstanceAttribute.fromColor(color)
+                }
+            }),
+            appearance : new PerInstanceColorAppearance({
+                translucent : false,
+                flat : true
+            }),
+            asynchronous : false
+        });
+    }
 
-            // Create the light
-            shadowMap._debugLight = new Primitive({
-                geometryInstances : new GeometryInstance({
-                    geometry : new SphereOutlineGeometry({
-                        radius : 1.0
-                    }),
-                    modelMatrix : modelMatrix,
-                    attributes : {
-                        color : ColorGeometryInstanceAttribute.fromColor(Color.BLUE)
-                    }
-                }),
-                appearance : new PerInstanceColorAppearance({
-                    translucent : false,
-                    flat : true
-                }),
-                asynchronous : false
-            });
+    var debugCascadeColors = [Color.RED, Color.GREEN, Color.BLUE, Color.MAGENTA];
 
-            //// Create the camera frustum
-            //var nearPlane = frameState.camera.frustum.near;
-            //var farPlane = frameState.camera.frustum.far;
-            //shadowMap._debugCameraFrustum = new Primitive({
-            //    geometryInstances : new GeometryInstance({
-            //        geometry : new BoxOutlineGeometry({
-            //            minimum : new Cartesian3(-0.0, -0.0, -nearPlane),
-            //            maximum : new Cartesian3(0.0, 0.0, -farPlane)
-            //        }),
-            //        attributes : {
-            //            color : ColorGeometryInstanceAttribute.fromColor(Color.RED)
-            //        }
-            //    }),
-            //    appearance : new PerInstanceColorAppearance({
-            //        translucent : false,
-            //        flat : true
-            //    }),
-            //    asynchronous : false
-            //});
+    function applyDebugSettings(shadowMap, frameState) {
+        if (!shadowMap.debugShow) {
+            return;
         }
 
-        //if (!shadowMap.debugFreezeFrame) {
-        //
-        //    //var nearPlane = frameState.camera.frustum.near;
-        //    //var point = Matrix4.multiplyByPoint(frameState.context.uniformState.inverseViewProjection, new Cartesian3(0, 0, -nearPlane), new Cartesian3());
-        //    //console.log(point.x, point.y, point.z);
-        //    //console.log(frameState.camera.positionWC.x, frameState.camera.positionWC.y, frameState.camera.positionWC.z);
-        //    //
-        //    //// TODO : possibly get this from the camera instead
-        //    //var cameraMatrix = frameState.context.uniformState.inverseViewProjection;
-        //    //shadowMap._debugCameraFrustum.modelMatrix = cameraMatrix;
-        //}
-
+        shadowMap._debugLightFrustum = shadowMap._debugLightFrustum || createDebugFrustum(Color.YELLOW);
```

Comment:
Even better is to use `defaultValue`, but since that can't be short-circuited, the best option here would be:

```javascript
var debugLightFrustum = shadowMap._debugLightFrustum;
if(!defined(debugLightFrustum)) {
    shadowMap._debugLightFrustum = createDebugFrustum(Color.YELLOW);
}
```